tests/common/helpers/parallel.py

```diff
@@ -104,6 +103,34 @@ def force_terminate(workers):
                         """Processes running target "{}" could not be terminated.
                         Unable to kill {}:{}, error:{}""".format(target.__name__, p.pid, p.name, err)
                     )
+                finally:
+                    # If sanity check is timeout and killed, we still want to put some information in results.
+                    # process name looks like _check_bgp_on_dut--<MultiAsicSonicHost sonic>
+                    # parse it to get sanity check item and hostname
+                    # For other processes, just return a dict {'failed': True}.
+                    # node looks like <EosHost VM0100> or <MultiAsicSonicHost sonic>
+                    node = p.name.split('--')[1]
+                    node_items = node.split()
+                    if len(node_items) > 1:
+                        node_name = node_items[1][:-1]
+                    else:
+                        node_name = node_items[0]
+                    function_name = p.name.split('--')[0]
+                    # sanity check funtions:
+                    # _check_interfaces_on_dut;_check_bgp_on_dut;_check_dbmemory_on_dut;_check_monit_on_dut;_check_processes_on_dut
+                    # parse function name to check item
+                    function_items = function_name.split('_')
+                    if len(function_items) > 2:
+                        check_item_name = function_items[2]
+                        if check_item_name in ['interfaces', 'bgp', 'processes', 'monit', 'dbmemory']:
+                            check_result = {"failed": True, "check_item": check_item_name, "host": node_name}
+                            results[node_name] = check_result
```

Comment:
This parallel_run is a common function used not just by sanity check. I don't think it is a good idea to add some sanity check specific logic here.

Reply:
@wangxin Per discussion, will submit a new PR to cover this part.